### PR TITLE
bpo-44337: Port LOAD_ATTR to PEP 659 adaptive interpreter

### PR DESCRIPTION
**Pull request by markshannon**
Implements four specializations of `LOAD_ATTR`:

* `LOAD_ATTR_SLOT` for slots.
* `LOAD_ATTR_SPLIT_KEYS` for attributes in instance dictionaries that have split keys.
* `LOAD_ATTR_HINT` for attributes in instance dictionaries that do not  have split keys.
* `LOAD_ATTR_MODULE` for module attributes

There are more specializations of `LOAD_ATTR` that we want to implement, but this PR is large enough already.

https://bugs.python.org/issue44337

### CODE REVIEW

**Review by methane — APPROVED** [review]

**Comment by pablogsal** [comment]
@markshannon You may want to check out https://bugs.python.org/msg395432

**Comment by markshannon** [comment]
Hopefully https://github.com/python/cpython/pull/26626 will fix the address sanitizer failures

File: Python/ceval.c [hunk 1]
```diff
@@ -3443,196 +3452,129 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, PyFrameObject *f, int throwflag)
...
+            assert(tp->tp_dictoffset > 0);
+            PyDictObject *dict = *(PyDictObject **)(((char *)owner) + tp->tp_dictoffset);
+            DEOPT_IF(dict == NULL, LOAD_ATTR);
+            assert(PyDict_CheckExact((PyObject *)dict));
+            DEOPT_IF(dict->ma_keys->dk_version != cache1->dk_version_or_hint, LOAD_ATTR);
+            res = dict->ma_values[cache0->index];
+            DEOPT_IF(res == NULL, LOAD_ATTR);
+            STAT_INC(loadattr_hit);
+            record_cache_hit(cache0);
+            STAT_INC(loadattr_hit);
```
Comment (cfbolz) [inline Python/ceval.c:3504]: Just casually reading through some of the code, it looks like the `STAT_INC(loadattr_hit);` is duplicated here (see two lines up).

Comment (markshannon) [reply Python/ceval.c:3504]: Thanks Carl. https://github.com/python/cpython/pull/26718

**Pull request merged**